pkg/xds/bootstrap/template_v3.go

```diff
@@ -326,3 +313,58 @@ func clusterTypeFromHost(host string) envoy_cluster_v3.Cluster_DiscoveryType {
 	}
 	return envoy_cluster_v3.Cluster_STRICT_DNS
 }
+
+func getGrpcServices(params configParameters, useTokenPath bool) []*envoy_core_v3.GrpcService {
```

Comment:
get from where? 🤔 It's more of a `buildGrpcServices` or just `grpcServices`. Also, shouldn't this return `*envoy_core_v3.GrpcService` instead of the list since it's always only one element?